wgsl/index.bs

```diff
@@ -2556,42 +2556,90 @@ pipeline_stage
 
 Each shader stage has its own set of features and constraints, described elsewhere.
 
-## Entry point declaration TODO ## {#entry-point-decl}
+## Entry point declaration ## {#entry-point-decl}
 
-The `entry_point` declares an entry point into the module. The entry points may
-be forward declarations but the functions referenced must be declared in the
-file.
+An <dfn noexport>entry point</dfn> is a function that is invoked to perform
+the work for a particular [=shader stage=].
 
-The input and output parameters to the entry point are determined by which
-global variables are used in the function and any called functions.
+To declare an entry point, write the name of the shader stage before the
+function declaration.
+
+When configuring the stage in the pipeline, the entry point is specified by providing
+the [SHORTNAME] module and the entry point's function name.
 
-The entry point function must be declared with no arguments and returning `void`.
+An entry point function must have no parameters,
+and its return type must be [=void=].
+
+ISSUE: We've lost the ability to use shader names that are invalid identifiers.
 
 <pre class='def'>
 entry_point_decl:
-   : ENTRY_POINT pipeline_stage EQUAL IDENT
-   | ENTRY_POINT pipeline_stage AS STRING_LITERAL EQUAL IDENT
-   | ENTRY_POINT pipeline_stage AS IDENT EQUAL IDENT
+   : entry_point_decoration_decl? pipeline_stage function_decl
+
+entry_point_decoration_decl
+  : ATTR_LEFT entry_point_decoration ATTR_RIGHT
 </pre>
 
 <div class='example' heading='Entry Point'>
   <xmp>
-    entry_point vertex = main
-       OpEntryPoint Vertex %vtx_main "vtx_main" %gl_FragColor
+    vertex fn vtx_main() -> void { gl_Position = vec4<f32>(); }
```

Comment:
did we consider making the stage selection more... regular in terms of syntax?
E.g. `fn<vertex> vtx_main() -> void {... }`

Reply:
We didn't discuss it much.
I didn't even know if the stage name should go before or after fn.

I like the template syntax you suggested.

Reply:
This is some rework of the grammar, so I'd like to agree on this in the group before making this change.

Reply:
Resolution in [2020-09-15](https://docs.google.com/document/d/1FbHuoLT7Pmklgjx4_09KOCOFHQGwrfVXGvaoGytIOsU/edit#heading=h.22uy73bqa3qv) meeting is to use syntax like this:

        [[stage(fragment)]]  fn main()....